File: c/resistor-color/src/resistor_color.c
#include "resistor_color.h"

resistor_band_t color_code(resistor_band_t band) {
    return band;
}

resistor_band_t* colors(void) {
    static resistor_band_t _colors[] = {
        BLACK, BROWN, RED, ORANGE, YELLOW,
        GREEN, BLUE, VIOLET, GREY, WHITE
    };

    return _colors;
}

File: java/resistor-color/src/main/java/ResistorColor.java
import java.util.Arrays;
import java.util.ArrayList;

class ResistorColor {
    static ArrayList<String> colors = new ArrayList<String>(Arrays.asList(
        "black", "brown", "red", "orange", "yellow",
        "green", "blue", "violet", "grey", "white"
    ));

    int colorCode(String color) {
        return colors.indexOf(color);
    }

    String[] colors() {
        return colors.toArray(new String[colors.size()]);
    }
}
